### PR TITLE
[5/N] Fix clang-tidy warnings in aten/src/ATen 

### PR DESCRIPTION
**Pull request by cyyever**
Follows #132001

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: aten/src/ATen/native/cpu/FlashAttentionKernel.cpp [hunk 1]
```diff
@@ -654,7 +654,11 @@ void cpu_flash_attention_backward(
               fill_stub(row_ptr + last_col + 1, static_cast<accum_t>(0), kvBlockSize - last_col - 1);
             }
           }
+#ifdef _MSC_VER
           if (is_reduced_type) {
```
Comment (Skylion007) [inline aten/src/ATen/native/cpu/FlashAttentionKernel.cpp:658]: I think we have a CONSTEXPR IF NOT MSVC macro somewhere for this.

Comment (Skylion007) [reply aten/src/ATen/native/cpu/FlashAttentionKernel.cpp:658]: I was thinking of CONSTEXPR_EXCEPT_WIN_CUDA, but I wonder if have another macro for this.

Comment (cyyever) [reply aten/src/ATen/native/cpu/FlashAttentionKernel.cpp:658]: It should be a MSVC2019 bug, however, there is only CONSTEXPR_EXCEPT_WIN_CUDA in c10/macros/Macros.h but this is not a CUDA file.

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**